### PR TITLE
Add multiple --prepare support

### PR DESCRIPTION
**Pull request by iamsauravsharma**
Add support for multiple --prepare option. Corresponding --prepare option will be used for command for benchmarking according to position. If --prepare option is less than benchmark command then all other remaining benchmark command will use last prepare option ( This logic may not be best logic to implement out). If prepare option is more than command for benchmarking then extra --prepare option will not be used
Part of #216

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you for your contribution.

> If --prepare option is less than benchmark command then all other remaining benchmark command will use last prepare option ( This logic may not be best logic to implement out). If prepare option is more than command for benchmarking then extra --prepare option will not be used

I would solve both problems by issuing an error message if `--prepare` has not been specified exactly once or exactly *N* times. This way, we don't have to make these decisions. Either, we use one preparation-command for all benchmark-commands, or we have a 1:1 mapping of preparation-commands to benchmark-commands.

**Comment by iamsauravsharma** [comment]
@sharkdp How would you like to raise error currently I am directly raising error by calling error function which is simple. And any suggestion over error message which was raised I think this is not best error message for this error

File: src/hyperfine/app.rs [hunk 1]
```diff
@@ -76,10 +76,14 @@ fn build_app() -> App<'static, 'static> {
...
+                .multiple(true)
+                .number_of_values(1)
                 .value_name("CMD")
                 .help(
-                    "Execute CMD before each timing run. This is useful for \
-                     clearing disk caches, for example.",
+                    "Execute CMD before each timing run. For example, this is \
+                     useful for clearing disk caches. This can be mentioned \
+                     out multiple time for each command or only one time for \
+                     all command.",
```
Comment (sharkdp) [inline src/hyperfine/app.rs]: Can we please rephrase this to:

Execute CMD before each timing run. This is useful for clearing disk caches, for example.
The `--prepare` option can be specified once for all commands or multiple times, once
for each command. In the latter case, each preparation command will be run prior to the
corresponding benchmark command.

File: src/main.rs [hunk 2]
```diff
@@ -113,7 +123,11 @@ fn build_hyperfine_options(matches: &ArgMatches<'_>) -> Result<HyperfineOptions,
         (None, None) => {}
     };
 
-    options.preparation_command = matches.value_of("prepare").map(String::from);
+    if let Some(values) = matches.values_of("prepare") {
+        options.preparation_command = Some(values.map(String::from).collect::<Vec<String>>());
+    } else {
+        options.preparation_command = None;
+    }
```
Comment (sharkdp) [inline src/main.rs]: I think this could be written as (untested):
``` rust
options.preparation_command = matches.values_of("prepare").map(|values| {
  values.map(String::from).collect::<Vec<_>>()
});
```

File: src/hyperfine/benchmark.rs [hunk 3]
```diff
@@ -234,13 +234,14 @@ pub fn run_benchmark(
...
-            Some((param, value)) => Command::new_parametrized(preparation_command, param, value),
-            None => Command::new(preparation_command),
-        });
+    let mut prepare_cmd = None;
+    if let Some(values) = &options.preparation_command {
+        if values.len() == 1 {
+            prepare_cmd = Some(new_command_from_get_parameter(cmd, &values[0]));
+        } else {
+            prepare_cmd = Some(new_command_from_get_parameter(cmd, &values[num]));
+        }
```
Comment (sharkdp) [inline src/hyperfine/benchmark.rs]: how about first storing `&values[0]` or `&values[num]` in a local variable. Then, you only need to call `new_command_from…` once. If this works, we could remove that function again and inline it here.

File: src/hyperfine/benchmark.rs [hunk 4]
```diff
@@ -234,13 +234,14 @@ pub fn run_benchmark(
...
 
     // Run init command
-    let prepare_cmd = options
-        .preparation_command
-        .as_ref()
-        .map(|preparation_command| match cmd.get_parameter() {
-            Some((param, value)) => Command::new_parametrized(preparation_command, param, value),
-            None => Command::new(preparation_command),
-        });
+    let mut prepare_cmd = None;
```
Comment (sharkdp) [inline src/hyperfine/benchmark.rs]: instead of using `mut` here, I would prefer
``` rust
let prepare_cmd = if … {
  …
} else {
  None
};
```
or (even better) a `.map` over the `options.preparation_command`, like it was before.
